src/nbl/video/CVulkanLogicalDevice.cpp

```diff
@@ -11,11 +11,10 @@
 using namespace nbl;
 using namespace nbl::video;
 
-
-
 CVulkanLogicalDevice::CVulkanLogicalDevice(core::smart_refctd_ptr<const IAPIConnection>&& api, renderdoc_api_t* const rdoc, const IPhysicalDevice* const physicalDevice, const VkDevice vkdev, const SCreationParams& params)
 : ILogicalDevice(std::move(api),physicalDevice,params,rdoc),
-m_vkdev(vkdev), m_devf(vkdev), m_deferred_op_mempool(NODES_PER_BLOCK_DEFERRED_OP*sizeof(CVulkanDeferredOperation),1u,MAX_BLOCK_COUNT_DEFERRED_OP,static_cast<uint32_t>(sizeof(CVulkanDeferredOperation)))
+m_vkdev(vkdev), m_devf(vkdev), m_deferred_op_mempool(NODES_PER_BLOCK_DEFERRED_OP*sizeof(CVulkanDeferredOperation),1u,MAX_BLOCK_COUNT_DEFERRED_OP,static_cast<uint32_t>(sizeof(CVulkanDeferredOperation))),
+m_spirvDebloater(core::make_smart_refctd_ptr<asset::ISPIRVDebloater>())
```

Comment:
make the debloater live in `ILogicalDevice` because I want you to give the same treatment to SPIR-V modules which are inputs to pipeline creation for any potential future backend